### PR TITLE
Monster: Smoky bear

### PR DESCRIPTION
**Pull request by Frost-wood**
This monster is done more on a whim after seeing comments on the df forums of smokey the bear.  This bear's stats are almost similar to the regular bear(except a little slower), but the bear when it moves creates a fire in a + or x pattern.  To prevent random base-destroying the bear automatically extinguishes the fire every turn.  However if you kill the bear on the wrong turn(or someone else does) before the bear extinguishes it then you might have a problem.

This bear is more a psychological threat to a mid-level player, as in testing a level 3-5 player with low (6-8)stats can easily take down the bear.  Even through the bear is mostly harmless the prospect of random fire will probably alarm the player into taking down the bear.

### CODE REVIEW

**Comment by vache** [comment]
I would rather just have it extinguish nearby fires every turn.  That's less in your face gimicky than making fire trails, and it gives it a bit of difficulty by eliminating the "kite into a bush and set it on fire" strat.

**Comment by Coolthulhu** [comment]
Pretty sure it should have an entry in "no joke monsters" mod.

**Comment by Rivet-the-Zombie** [comment]
Yeah this is definitely a joke thing. Monsters breathing (or otherwise excreting) fire is a bit on the fantasy side of things for mainline DDA.

**Comment by kevingranade** [comment]
The problem with it putting out fires is it will hardly ever happen.
Possibly make it super agressive if it sees the player use fire?  That's
still a stretch.
That having been said though, that kind of leaves it a very confused
monster?  If it's given up on stopping fires and decided to be a firebug,
why put the fires out?

**Comment by Rivet-the-Zombie** [comment]
I'm more bothered by this critter's ability to magically produce and extinguish fire at will than I am by its behavior.

If DDA were set in Middle Earth this would be vaguely justifiable as some sort of magic creature, but in DDA it just makes no sense.

**Comment by Frost-wood** [comment]
I was afraid it might be a little too fantastical, but I thought it might be worth a try.

**Comment by Rivet-the-Zombie** [comment]
How about instead of a magical fire-bear, just make it a smoker zed/zombear type thing; a special mutant zombear that emits a cloud of smoke like the extant smoke-zeds? It would fit the lore readily enough and still be a bit of a pun, while still being serious enough that it could be a new form of special zombie.

**Comment by Chezzo** [comment]
Be careful! Smokey has lawyers. http://readersupportednews.org/off-site-news-section/49-49/17314-focus-forest-service-seeks-to-silence-smokey-the-bear-over-fracking

**Comment by KA101** [comment]
Yeah, when I saw the title I presumed smoker zombear.  That'd be acceptable.

Now, if the original showed up around hellmouths...

*KA101 has considered firebats and firefoxes as potential Nether critters*

Both would be from Nether heat dimensions (cf. the ice labs resulting from Nether cold dimensions) with the firefox being a joke critter, Friendly!, curious, cute/morale-boosting, light-producing, and unfortunately not realizing that folks/stuff on this plane doesn't react well to burning.  (It'd start fires on flammable tiles, but you'd only ever meet one underground, likely on non-flammable flooring.  And then you take your new pet home...)

**Comment by Frost-wood** [comment]
Fixed the bear to be a regular smoker bear, removed firetrail and extinguish, but kept fireproof.  Ironically the "FIREY" tag doesn't prevent damage from fire or set objects on fire, but prevents you from being set on fire.

File: data/json/monstergroups.json [hunk 1]
```diff
@@ -334,7 +334,8 @@
       { "monster" : "mon_worm", "freq" :  1, "cost_multiplier" : 0, "starts" : 168, "conditions" : ["SPRING", "SUMMER", "AUTUMN"] },
       { "monster" : "mon_worm", "freq" :  1, "cost_multiplier" : 0, "starts" : 336, "conditions" : ["SPRING", "SUMMER", "AUTUMN"] },
       { "monster" : "mon_worm", "freq" :  1, "cost_multiplier" : 0, "starts" : 504, "conditions" : ["SPRING", "SUMMER", "AUTUMN"] },
-      { "monster" : "mon_worm", "freq" :  1, "cost_multiplier" : 0, "starts" : 672, "conditions" : ["SPRING", "SUMMER", "AUTUMN"] }
+      { "monster" : "mon_worm", "freq" :  1, "cost_multiplier" : 0, "starts" : 672, "conditions" : ["SPRING", "SUMMER", "AUTUMN"] },
+	  { "monster" : "mon_bear_smoky", "freq" :  1, "cost_multiplier" : 0, "starts" : 672, "conditions" : ["SPRING", "SUMMER", "AUTUMN"] }
```
Comment (KA101) [inline data/json/monstergroups.json]: Tabs and this is OK for mergetest.

**Comment by Frost-wood** [comment]
hmm the tabs appear perfectly alright in notepad++ but when I upload it to git it appears different, and I don't know how to fix it.

**Comment by Rivet-the-Zombie** [comment]
Don't use tabs (like, ever, in .json) use spaces.

**Comment by Frost-wood** [comment]
Thanks!

**Comment by DavidKeaton** [comment]
There should be an option for both soft tabs and tab expansion.

File: data/json/monsters.json [hunk 2]
```diff
@@ -122,6 +122,38 @@
...
+    "melee_cut":4,
+    "dodge":3,
+    "armor_bash":4,
+    "armor_cut":4,
+    "vision_day":30,
+    "vision_night":10,
+    "luminance":0,
+    "hp":100,
+    "death_function":"SMOKEBURST",
+    "description":"Smoke bellows from this black-white bear's fur, and the ground bursts into flame in its wake, only to be put out a moment later.",
```
Comment (Rivet-the-Zombie) [inline data/json/monsters.json]: Fix this description to reflect that it no longer has fire magic, and this should be ready for merge testing.

**Pull request merged**